net/redistest/redistest.go

```diff
@@ -2,15 +2,29 @@ package redistest
 
 import (
 	"context"
+	"net"
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/docker/go-connections/nat"
 	"github.com/redis/go-redis/v9"
 	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/network"
 	"github.com/testcontainers/testcontainers-go/wait"
 )
 
+// testNetwork returns a network for test containers.
+// Containers on this network will use the same port and different addresses
+// unlike containers on the host network, which use different ports and the same loopback address.
+// The network will be cleaned up automatically by testcontainers at exit.
+var testNetwork = sync.OnceValues(func() (*testcontainers.DockerNetwork, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	return network.New(ctx)
+})
```

Comment:
This does not pass noleak :(
```
noleak: 1 active

goroutine 12800 [chan receive]:
github.com/testcontainers/testcontainers-go.(*Reaper).Connect.func1({0x1b4f8f8, 0xc000526650})
	/home/runner/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.27.0/reaper.go:360 +0x76e
created by github.com/testcontainers/testcontainers-go.(*Reaper).Connect in goroutine 12769
	/home/runner/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.27.0/reaper.go:325 +0x170
```
Need to call network.Remove() after last container stops using the network.

Reply:
Fixed